src/middleware/middleware.js

```diff
@@ -28,7 +28,7 @@ export default function ({ store, query, route }) {
     })
   }
 
-  if (store.state.nav.isReferredFromCc) {
+  if (!process.server && store.state.nav.isReferredFromCc) {
```

Comment:
nit: FWIW positive assertions always seem easier for me to quickly grok than a negation. Clearer for new folks or someone skimming as well who might not have the full context of `process.server` and `process.client`.

```suggestion
  if (process.client && store.state.nav.isReferredFromCc) {
```

Reply:
This is classic half-knowledge stuff. I didn't even know there was a `process.client`!